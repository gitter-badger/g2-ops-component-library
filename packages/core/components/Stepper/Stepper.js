// @flow

import type { Node } from 'react'

import React from 'react'
import { Stepper as MuiStepper, Step, StepLabel } from 'material-ui/Stepper'

import { wrapMuiContext } from '../../wrapMuiContext'

const Stepper =  wrapMuiContext(MuiStepper)
export { Stepper, Step, StepLabel }
