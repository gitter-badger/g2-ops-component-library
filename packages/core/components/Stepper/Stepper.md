Stepper Example:

```js
var StepperExample = require('../../examples/StepperExample.js').default;
<StepperExample />
```

For more documentation on usage: http://www.material-ui.com/#/components/stepper
