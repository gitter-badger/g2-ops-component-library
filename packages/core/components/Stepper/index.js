export { Stepper, Step, StepLabel } from './Stepper'
export default from './Stepper'
