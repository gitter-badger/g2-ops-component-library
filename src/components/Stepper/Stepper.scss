.stepper > div > div > span {
  flex-direction: column;
  font-family: 'Roboto', sans-serif !important;
}
