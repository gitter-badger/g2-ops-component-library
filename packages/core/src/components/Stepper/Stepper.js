// @flow

import type { Node } from 'react'

import React from 'react'
import { Stepper as MuiStepper, Step, StepLabel } from 'material-ui/Stepper'

import { wrapMuiContext } from 'utilities/wrapMuiContext'
import './Stepper.scss'

export const Stepper = wrapMuiContext(<T: { children: Node }>(props: T) => {
  return (
    <div className="g2-ops-stepper">
      <MuiStepper {...props}>{props.children}</MuiStepper>
    </div>
  )
})

export { Step, StepLabel }
