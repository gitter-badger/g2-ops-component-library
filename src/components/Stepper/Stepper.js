import React from 'react'
import { Stepper as MuiStepper, Step, StepLabel } from 'material-ui/Stepper'
import { wrapMuiContext } from '../../wrapMuiContext'

/**
 *
 * @example ../../examples/Stepper.md
 */
const StepperComponent = (props) => wrapMuiContext(
  <MuiStepper {...props}>
    {props.children}
  </MuiStepper>
)


export const Stepper = wrapMuiContext(StepperComponent)
export { Step, StepLabel }
