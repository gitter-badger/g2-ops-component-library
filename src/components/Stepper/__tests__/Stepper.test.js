import React from 'react'
import renderer from 'react-test-renderer'

import Stepper, { Step, StepLabel } from '../Stepper'

describe('<Stepper />', () => {
  test('should render properly', () => {
    const stepIndex = 1
    const tree = renderer.create(
      <Stepper activeStep={stepIndex}>
        <Step>
          <StepLabel>Select campaign settings</StepLabel>
        </Step>
        <Step>
          <StepLabel>Create an ad group</StepLabel>
        </Step>
        <Step>
          <StepLabel>Create an ad</StepLabel>
        </Step>
      </Stepper>,
    )
    expect(tree).toMatchSnapshot()
  })
})
