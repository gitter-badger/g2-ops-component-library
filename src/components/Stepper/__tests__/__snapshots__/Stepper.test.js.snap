// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stepper /> should render properly 1`] = `
<div
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "muiPrepared": true,
    }
  }
>
  <div
    style={
      Object {
        "flex": "0 0 auto",
        "muiPrepared": true,
      }
    }
  >
    <span
      disabled={undefined}
      style={
        Object {
          "alignItems": "center",
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "flex",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 14,
          "height": 72,
          "muiPrepared": true,
          "paddingLeft": 14,
          "paddingRight": 14,
        }
      }
    >
      <span
        style={
          Object {
            "muiPrepared": true,
            "paddingRight": 8,
          }
        }
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#1d5ab9",
              "display": "block",
              "fill": "#1d5ab9",
              "fontSize": 24,
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
      </span>
      Select campaign settings
    </span>
  </div>
  <div
    style={
      Object {
        "flex": "1 1 auto",
        "muiPrepared": true,
      }
    }
  >
    <span
      style={
        Object {
          "borderColor": "#bdbdbd",
          "borderTopStyle": "solid",
          "borderTopWidth": 1,
          "display": "block",
          "marginLeft": -6,
          "muiPrepared": true,
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "flex": "0 0 auto",
        "marginLeft": -6,
        "muiPrepared": true,
      }
    }
  >
    <span
      disabled={undefined}
      style={
        Object {
          "alignItems": "center",
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "flex",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 14,
          "fontWeight": 500,
          "height": 72,
          "muiPrepared": true,
          "paddingLeft": 14,
          "paddingRight": 14,
        }
      }
    >
      <span
        style={
          Object {
            "muiPrepared": true,
            "paddingRight": 8,
          }
        }
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#1d5ab9",
              "display": "block",
              "fill": "#1d5ab9",
              "fontSize": 24,
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <text
            fill="#fff"
            fontSize="12"
            textAnchor="middle"
            x="12"
            y="16"
          >
            2
          </text>
        </svg>
      </span>
      Create an ad group
    </span>
  </div>
  <div
    style={
      Object {
        "flex": "1 1 auto",
        "muiPrepared": true,
      }
    }
  >
    <span
      style={
        Object {
          "borderColor": "#bdbdbd",
          "borderTopStyle": "solid",
          "borderTopWidth": 1,
          "display": "block",
          "marginLeft": -6,
          "muiPrepared": true,
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "flex": "0 0 auto",
        "marginLeft": -6,
        "muiPrepared": true,
      }
    }
  >
    <span
      disabled={true}
      style={
        Object {
          "alignItems": "center",
          "color": "rgba(0, 0, 0, 0.26)",
          "cursor": "default",
          "display": "flex",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 14,
          "height": 72,
          "muiPrepared": true,
          "paddingLeft": 14,
          "paddingRight": 14,
        }
      }
    >
      <span
        style={
          Object {
            "muiPrepared": true,
            "paddingRight": 8,
          }
        }
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#9e9e9e",
              "display": "block",
              "fill": "#9e9e9e",
              "fontSize": 24,
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <text
            fill="#fff"
            fontSize="12"
            textAnchor="middle"
            x="12"
            y="16"
          >
            3
          </text>
        </svg>
      </span>
      Create an ad
    </span>
  </div>
</div>
`;
